src/plugins/irc/irc-protocol.c

```diff
@@ -292,6 +292,36 @@ IRC_PROTOCOL_CALLBACK(away)
 }
 
 /*
+ * Callback for the IRC message "ACCOUNT": account info about a nick
+ * (with capability "account-notify").
+ *
+ * Message looks like:
+ *   :nick!user@host ACCOUNT *
+ *   :nick!user@host ACCOUNT accountname
+ */
+
+IRC_PROTOCOL_CALLBACK(account)
+{
+    struct t_irc_channel *ptr_channel;
+    struct t_irc_nick *ptr_nick;
+
+    IRC_PROTOCOL_MIN_ARGS(3);
+
+    for (ptr_channel = server->channels; ptr_channel;
+         ptr_channel = ptr_channel->next_channel)
+    {
+        ptr_nick = irc_nick_search (server, ptr_channel, nick);
+        if (ptr_nick)
+        {
+            free (ptr_nick->account);
```

Comment:
I would check the value of `ptr_nick->account` here, just to be sure it's set in nick, ie:
```c
if (ptr_nick->account)
    free (ptr_nick->account)
```